crates/bevy_ecs/macros/src/lib.rs

```diff
@@ -45,9 +45,9 @@ impl Parse for AllTuples {
 #[proc_macro]
 pub fn all_tuples(input: TokenStream) -> TokenStream {
     let input = parse_macro_input!(input as AllTuples);
-    let len = input.end - input.start;
+    let len = input.end - input.start + 1;
```

Comment:
I'd change the input to be in format `start..=end`, to make it clearer

Reply:
On the other hand, up to 15 inputs almost feels nicer to me. Not sure about that though. So long as you use an explicit range syntax as the surface syntax to make it clear

Reply:
changed to a range. 16 is to keep the same limitation as 0.4, no hard preference on my side